phpunit/class-core-block-theme-test.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Core block theme tests.
+ *
+ * @package Gutenberg
+ */
+
+/**
+ * Test inclusion of opt-in core block theme.
+ */
+class Core_Block_Theme_Test extends WP_UnitTestCase {
+	private $old_wp_styles;
+	private $old_wp_scripts;
+
+	function setUp() {
+		parent::setUp();
+
+		$this->old_wp_scripts = isset( $GLOBALS['wp_scripts'] ) ? $GLOBALS['wp_scripts'] : null;
+		remove_action( 'wp_default_scripts', 'wp_default_scripts' );
+
+		$GLOBALS['wp_scripts']                  = new WP_Scripts();
+		$GLOBALS['wp_scripts']->default_version = get_bloginfo( 'version' );
+
+		$this->old_wp_styles = isset( $GLOBALS['wp_styles'] ) ? $GLOBALS['wp_styles'] : null;
+		remove_action( 'wp_default_styles', 'wp_default_styles' );
+
+		$GLOBALS['wp_styles']                  = new WP_Styles();
+		$GLOBALS['wp_styles']->default_version = get_bloginfo( 'version' );
+	}
+
+	function tearDown() {
+		$GLOBALS['wp_scripts'] = $this->old_wp_scripts;
+		add_action( 'wp_default_scripts', 'wp_default_scripts' );
+
+		$GLOBALS['wp_styles'] = $this->old_wp_styles;
+		add_action( 'wp_default_styles', 'wp_default_styles' );
+
+		if ( current_theme_supports( 'wp-block-styles' ) ) {
+			remove_theme_support( 'wp-block-styles' );
+		}
+
+		parent::tearDown();
+	}
+
+	function test_block_theme_in_editor_without_theme_support() {
```

Comment:
I think you could add a comment to (at least) this test (and maybe the next) stating default block styles are always loaded for the editor and why. I know you've explained it elsewhere in this PR but that documentation won't be obvious to someone reading this test/just this file, so it bears repeating.

It's reasonably obvious what these tests are doing but not obvious *why*. I think a bit more explanation would make them clearer.

Reply:
Agreed. I'll make the update.